src/nvim/popupmenu.c

```diff
@@ -100,6 +100,36 @@ static void pum_compute_size(void)
   }
 }
 
+static int pum_redr_border(void)
```

Comment:
if possible we should reuse most of `win_redr_border` . It is only called in one place, so the window specific parts could be moved out and instead have something like `grid_draw_border` which takes `grid` and inner height/width as parameters.